client_test.go

```diff
@@ -163,6 +163,53 @@ func getHostAndPortFromTestServer(ts *httptest.Server) (string, int, error) {
 	return hostSplits[0], port, nil
 }
 
+func TestDeleteDatasetExisting(t *testing.T) {
+	assert := assert.New(t)
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal("DELETE", r.Method)
+		assert.Equal("/v1/configuration/datasets/uuid1", r.URL.Path)
+		w.Header().Set("Content-Type", "application/json")
+		w.Write([]byte("This is an example server.\n"))
+		b, err := json.Marshal(configurationPayload{
+			DatasetID: "uuid1",
+			Primary:   "primary1",
+			Deleted:   true,
+		})
+		assert.NoError(err)
+		w.Write(b)
+	},
+	))
+
+	host, port, err := getHostAndPortFromTestServer(ts)
+	assert.NoError(err)
+
+	c := newFlockerTestClient(host, port)
+	assert.NoError(err)
+
+	c.DeleteDataset("uuid1")
+	assert.NoError(err)
```

Comment:
The assertion is fixed. The other problem is something I copy pasted from another test.